check_modulemd.py

```diff
@@ -209,12 +210,25 @@ def test_component_availability(self):
         """
         Are all the components we reference in the packages section available?
         """
-        # verify that the specified ref (if any, defaults to master HEAD) is available in the
-        # specified repository (if any, defaults to Fedora dist-git).
-        self.log.warn("Not yet implemented")
+        # verify that the specified ref (if none, defaults to master HEAD) is available in the
+        # specified repository (if none, defaults to Fedora dist-git).
         for p in self.mmd.components.rpms.values():
-            self.log.warn(
-                "Need to check availability of component RPM %s" % p.name)
+            repository = "http://src.fedoraproject.org/cgit/rpms/%s.git" % p.name
+            if p.repository:
+                repository = p.repository
+            ref = "HEAD"
+            if p.ref:
+                ref = p.ref
+            patch_path = "/patch/?id=%s" % ref
+            #Use wget to check if patch exits, --spider just check if page exists, do not download it
+            cmd = "wget " + repository + patch_path + " --spider -O - > /dev/null 2>&1"
```

Comment:
Can I use urllib to get just HEAD from the url? I do not want to download the whole file.
httplib module might be an option for this.